File: problems/1.two_sum.cpp
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        int size = nums.size();
        std::vector<int> result(2);
        for (int i = 0; i < size; i++) {
            for (int j = i + 1; j < size; j++) {
                if (nums[i] + nums[j] == target) {
                    result[0] = i;
                    result[1] = j;
                    return result;
                }
            }
        }
    }
}

File: interview_top/1.two_sum.go
package interviewtop

func twoSum(nums []int, target int) []int {
	dict := make(map[int]int)
	for i, v := range nums {
		if j, ok := dict[target-v]; ok {
			return []int{j, i}
		}
		dict[v] = i
	}
	return nil
}
